services-custom/dynamodb-enhanced/src/test/java/software/amazon/awssdk/extensions/dynamodb/mappingclient/functionaltests/BasicCrudTest.java

```diff
@@ -418,6 +436,17 @@ public void deleteWithConditionThatFails() {
                                                .build());
     }
 
+    @Test
+    public void updateItemInterfaceToOperationSucceeds() {
+        Record record = new Record()
+            .setId("id-value")
+            .setSort("sort-value")
+            .setAttribute("one");
+
+        mappedTable.updateItem(Record.class, r -> r.item(record));
+        mappedTable.updateItem(UpdateItemEnhancedRequest.builder(Record.class).item(record).build());
+    }
+
```

Comment:
I think we should be able to make basic assertions here beyond 'it didn't blow up'.